assets/js/components/ProjectEditorTextField.js

```diff
@@ -0,0 +1,163 @@
+import $ from 'jquery'
+import { MDCTextField } from '@material/textfield'
+import { CustomTranslationApi } from '../api/CustomTranslationApi'
+
+export function ProjectEditorTextField (projectDescriptionCredits, programId, programSection, hasDefaultText, snackbar) {
+  const self = this
+  this.programId = programId
+  this.programSection = programSection
+  this.hasDefaultText = hasDefaultText
+  this.snackbar = snackbar
+  this.customTranslationApi = new CustomTranslationApi(programSection)
+
+  this.editText = $('#edit-' + this.programSection + '-text')
+  this.editTextError = $('#edit-' + this.programSection + '-text-error')
+  this.textLoadingSpinner = $('#edit-' + this.programSection + '-loading-spinner')
+  this.selectedLanguage = $('#edit-selected-language')
+
+  this.defaultSection = $('#' + this.programSection)
+  this.initialText = this.defaultSection.text().trim()
+
+  new MDCTextField(document.querySelector('#edit-' + this.programSection + '-mdc-text-field'))
+
+  this.translationSaved = projectDescriptionCredits.data('trans-translation-saved')
+  this.translationDeleted = projectDescriptionCredits.data('trans-translation-deleted')
+
+  this.pathEditName = projectDescriptionCredits.data('path-edit-program-name')
+  this.pathEditDescription = projectDescriptionCredits.data('path-edit-program-description')
+  this.pathEditCredits = projectDescriptionCredits.data('path-edit-program-credits')
+
+  this.show = () => {
```

Comment:
When i think of it, this `ProjectEditorTextField` is never independently shown or hidden. it is shown when the `ProjectEditor` is shown. So I think this function is more like a `setText`